### DISCUSSION TITLE
Block comment throws an error when parsing except when it is the first thing in the file

### DISCUSSION THREAD

**Discussion by lucypero** [tree-sitter/tree-sitter#796] [Q&A] [answered]
https://github.com/lucypero/tree-sitter-odin

I am making a tree-sitter grammar for Odin. I am parsing the block comments in `src/scanner.c`.

If you `tree-sitter generate && tree-sitter parse examples/comments.odin` you can see the error.

```
/*
comment
nested /* comment */

*/

 /* another one


*/

package main

```

```
(source_file [0, 0] - [12, 0]
  (block_comment [0, 0] - [4, 2])
  (ERROR [6, 1] - [9, 2]
    (ERROR [6, 1] - [6, 3])
    (identifier [6, 4] - [6, 11])
    (identifier [6, 12] - [6, 15])
    (ERROR [9, 0] - [9, 2]))
  (package_clause [11, 0] - [11, 12]
    (package_identifier [11, 8] - [11, 12])))
```

If the block comment is the very first thing in the file, it is parsed correctly. But if it's not, even if it's just a newline that is above it, it errors out. I would really appreciate some help. Thanks

**Comment by maxbrunsfeld** [comment]
Try running your parse with `-d` (logging to stderr) or `-D` (graphical logging, requires `graphviz` to be installed). Also, have you run run your parse under a debugger like `gdb` or `lldb` to check what your scanner is doing?

**Reply by lucypero** [reply, marked as answer]
> Try running your parse with `-d` (logging to stderr) or `-D` (graphical logging, requires `graphviz` to be installed). Also, have you run run your parse under a debugger like `gdb` or `lldb` to check what your scanner is doing?

So, someone just told me to add this to my `external_scanner_scan()` function:

```while (iswspace(lexer->lookahead)) lexer->advance(lexer, true);```

And now it seems to work. At least it can now parse that file I mentioned... It appears that whitespace was the culprit. I don't know why.

I don't think I can use `-D` on Windows... I've been using `-d` but it didn't help me. I haven't tried debugging using a debugger but I'll try next time something like this happens. Thank you.

**Reply by maxbrunsfeld** [reply]
Yeah, if there is whitespace in your source file, your external scanner will need to handle it.
